### PR TITLE
DE2363: Fixes for donation/payment existing methods bla bla bla

### PR DESCRIPTION
**Pull request by samueljmello**

### CODE REVIEW

File: src/app/summary/summary.component.ts [hunk 1]
```diff
@@ -40,29 +40,48 @@ export class SummaryComponent implements OnInit {
...
-          this.paymentService.postPayment(paymentDetail).subscribe(
-            info => this.handleSuccess(info),
-            innerError => this.handleInnerError(innerError)
-          );
-        },
-        error => this.handleOuterError(error)
-    );
-    return false;
+
+    let pymt_type = this.gift.paymentType === 'ach' ? 'bank' : 'cc';
```
Comment (mjvcallibrity) [inline src/app/summary/summary.component.ts]: Variable name doesn't follow style in rest of code. Consider changing to `paymentType`.

File: src/app/summary/summary.component.ts [hunk 2]
```diff
@@ -40,29 +40,48 @@ export class SummaryComponent implements OnInit {
...
+    let paymentDetails = new PaymentCallBody('', this.gift.amount, pymt_type, 'PAYMENT', this.gift.invoiceId);
+
+    // adding a new payment method (must use donor token to create saved payment method)
+    if (this.gift.isNewPaymentMethod()) {
```
Comment (mjvcallibrity) [inline src/app/summary/summary.component.ts]: Consider renaming this method to `isUsingNewPaymentMethod`. Current name suggests that the gift is a type of payment method.

File: src/app/summary/summary.component.ts [hunk 3]
```diff
@@ -40,29 +40,48 @@ export class SummaryComponent implements OnInit {
...
+      this.paymentService.makeApiDonorCall(this.gift.donor).subscribe(
+          value => this.postTransaction(paymentDetails),
+          error => this.handleOuterError()
+      );
+
+    // using an existing payment method (payment method already exists. auth token handles the payment)
+    } else if (this.gift.isExistingPaymentMethod()) {
```
Comment (mjvcallibrity) [inline src/app/summary/summary.component.ts]: Consider renaming this method to `isUsingExistingPaymentMethod`. Current name suggests that the gift is a type of payment method.

File: src/app/summary/summary.component.ts [hunk 4]
```diff
@@ -71,41 +90,60 @@ export class SummaryComponent implements OnInit {
 
     this.beginProcessing();
 
+    // one time gifts
     if (this.gift.isOneTimeGift()) {
       let pymt_type = this.gift.paymentType === 'ach' ? 'bank' : 'cc';
```
Comment (mjvcallibrity) [inline src/app/summary/summary.component.ts]: Consider changing to `paymentType` to conform to rest of variable assignment stylings.

File: src/app/summary/summary.component.ts [hunk 5]
```diff
@@ -71,41 +90,60 @@ export class SummaryComponent implements OnInit {
...
+        this.gift.fund.ProgramId.toString(),
+        this.gift.amount,
+        pymt_type,
+        'DONATION',
+        this.gift.invoiceId
+      );
 
-      if(this.gift.donor){
+      // adding a new payment method (must use donor token to create saved payment method)
+      if (this.gift.isNewPaymentMethod()) {
```
Comment (mjvcallibrity) [inline src/app/summary/summary.component.ts]: Consider renaming this method to `isUsingNewPaymentMethod`. Current name suggests that the gift is a type of payment method.

File: src/app/summary/summary.component.ts [hunk 6]
```diff
@@ -71,41 +90,60 @@ export class SummaryComponent implements OnInit {
...
-                  innerError => this.handleInnerError(innerError)
-              );
+              this.postTransaction(donationDetails);
             },
-            error => this.handleOuterError(error)
+            error => this.handleOuterError()
         );
+
+      // using an existing payment method (payment method already exists. auth token handles the payment)
+      } else if (this.gift.isExistingPaymentMethod()) {
```
Comment (mjvcallibrity) [inline src/app/summary/summary.component.ts]: Consider renaming this method to `isUsingExistingPaymentMethod`. Current name suggests that the gift is a type of payment method.

File: src/app/summary/summary.component.ts [hunk 7]
```diff
@@ -71,41 +90,60 @@ export class SummaryComponent implements OnInit {
...
       }
 
-    } else {
+    // recurring gifts
+    } else if (this.gift.isRecurringGift()) {
 
-      let userPmtInfo: CustomerBank | CustomerCard  = this.gift.userCc || this.gift.userBank;
-      let stripeMethodName: string = this.gift.userCc ? 'getCardInfoToken' : 'getBankInfoToken';
+      // must use a new paymenth method (can't use existing)
+      if (this.gift.isNewPaymentMethod()) {
```
Comment (mjvcallibrity) [inline src/app/summary/summary.component.ts]: Consider renaming this method to `isUsingNewPaymentMethod`. Current name suggests that the gift is a type of payment method.

File: src/app/summary/summary.component.ts (hunk 1 above)
Comment (vchtelmakh) [inline src/app/summary/summary.component.ts]: @mjvcallibrity This is the way it is because of parameter names on the API object.

Comment (samueljmello) [inline src/app/summary/summary.component.ts]: Correct. I can fix this, but it was already like this.

File: src/app/summary/summary.component.ts (hunk 7 above)
Comment (samueljmello) [inline src/app/summary/summary.component.ts]: Done.

File: src/app/summary/summary.component.ts (hunk 6 above)
Comment (samueljmello) [inline src/app/summary/summary.component.ts]: Done.

File: src/app/summary/summary.component.ts (hunk 5 above)
Comment (samueljmello) [inline src/app/summary/summary.component.ts]: Done.

File: src/app/summary/summary.component.ts (hunk 3 above)
Comment (samueljmello) [inline src/app/summary/summary.component.ts]: Done.

File: src/app/summary/summary.component.ts (hunk 2 above)
Comment (samueljmello) [inline src/app/summary/summary.component.ts]: Done.

File: src/app/summary/summary.component.ts (hunk 1 above)
Comment (samueljmello) [inline src/app/summary/summary.component.ts]: Done.

**Pull request merged**